cmd/registry/cmd/annotate_test.go

```diff
@@ -0,0 +1,180 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"testing"
+
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"github.com/google/go-cmp/cmp"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestAnnotate(t *testing.T) {
+	var err error
+
+	projectId := "annotate-test"
+	projectName := "projects/" + projectId
+	apiId := "sample"
+	apiName := projectName + "/apis/" + apiId
+	versionId := "1.0.0"
+	versionName := apiName + "/versions/" + versionId
+	specId := "openapi.json"
+	specName := versionName + "/specs/" + specId
+
+	// Create a registry client.
+	ctx := context.Background()
+	registryClient, err := connection.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("error creating client: %+v", err)
+	}
+	defer registryClient.Close()
+	// Clear the test project.
+	err = registryClient.DeleteProject(ctx, &rpc.DeleteProjectRequest{
+		Name: projectName,
+	})
+	if err != nil && status.Code(err) != codes.NotFound {
+		t.Fatalf("error deleting test project: %+v", err)
+	}
+	// Create the test project.
+	_, err = registryClient.CreateProject(ctx, &rpc.CreateProjectRequest{
+		ProjectId: projectId,
+		Project: &rpc.Project{
+			DisplayName: "Test",
+			Description: "A test catalog",
+		},
+	})
+	if err != nil {
+		t.Fatalf("error creating project %s", err)
+	}
+	// Create a sample api.
+	_, err = registryClient.CreateApi(ctx, &rpc.CreateApiRequest{
+		Parent: projectName,
+		ApiId:  apiId,
+		Api:    &rpc.Api{},
+	})
+	if err != nil {
+		t.Fatalf("error creating api %s", err)
+	}
+	// Create a sample version.
+	_, err = registryClient.CreateApiVersion(ctx, &rpc.CreateApiVersionRequest{
+		Parent:       apiName,
+		ApiVersionId: versionId,
+		ApiVersion:   &rpc.ApiVersion{},
+	})
+	if err != nil {
+		t.Fatalf("error creating version %s", err)
+	}
+	// Create a sample spec.
+	_, err = registryClient.CreateApiSpec(ctx, &rpc.CreateApiSpecRequest{
+		Parent:    versionName,
+		ApiSpecId: specId,
+		ApiSpec: &rpc.ApiSpec{
+			MimeType: "application/x.openapi;version=3.0.0",
+			Contents: []byte(`{"openapi": "3.0.0", "info": {"title": "test", "version": "v1"}, "paths": {}}`),
+		},
+	})
+	if err != nil {
+		t.Fatalf("error creating spec %s", err)
+	}
+
+	testCases := []struct {
+		comment  string
+		args     []string
+		expected map[string]string
+	}{
+		{"add some annotations",
+			[]string{"a=1", "b=2"},
+			map[string]string{"a": "1", "b": "2"}},
+		{"remove one annotation and overwrite the other",
+			[]string{"a=3", "b-", "--overwrite"},
+			map[string]string{"a": "3"}},
+		{"changing an annotation without --overwrite should be ignored",
+			[]string{"a=4"},
+			map[string]string{"a": "3"}},
```

Comment:
Nit: Adding field names to table-driven test cases can aid readability quite a bit.

Reply:
done - I think it's arguably helpful at this small scale, but I'd support this as an general style for all our tests, and we might also want to standardize on some field names (optional todo)